client/components/sites-dropdown/index.jsx

```diff
@@ -45,20 +42,23 @@ export default React.createClass( {
 	},
 
 	getInitialState() {
-		const primary = sites.getPrimary();
+		const primarySite = this.props.selectedSiteId
+			? sites.getSite( this.props.selectedSiteId ) 
+			: sites.getPrimary();
+
 		return {
-			selected: this.props.selected || primary && primary.slug
+			selectedSiteSlug: primarySite && primarySite.slug
```

Comment:
We still need the `siteSlug`, because `site-selector` component expect the slug as `selected` property.
Ideally, the second step would be to apply the same patch of this pull request to `site-selector`.